app/src/main/java/de/westnordost/streetcomplete/util/ktx/LocalDate.kt

```diff
@@ -1,19 +1,48 @@
+@file:Suppress("NOTHING_TO_INLINE")
+
 package de.westnordost.streetcomplete.util.ktx
 
-import java.time.Instant
-import java.time.LocalDate
-import java.time.Month
-import java.time.ZoneId
+import kotlinx.datetime.Clock
+import kotlinx.datetime.DateTimeUnit
+import kotlinx.datetime.Instant
+import kotlinx.datetime.LocalDate
+import kotlinx.datetime.LocalDateTime
+import kotlinx.datetime.Month
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.atStartOfDayIn
+import kotlinx.datetime.minus
+import kotlinx.datetime.plus
+import kotlinx.datetime.toInstant
+import kotlinx.datetime.toLocalDateTime
 
 fun LocalDate.toInstant(): Instant =
-    this.atStartOfDay(ZoneId.systemDefault()).toInstant()
+    this.atStartOfDayIn(TimeZone.currentSystemDefault())
 
-fun LocalDate.toEpochMilli(): Long = this.toInstant().toEpochMilli()
+fun LocalDate.toEpochMilli(): Long = this.toInstant().toEpochMilliseconds()
 
 fun Instant.toLocalDate(): LocalDate =
-    this.atZone(ZoneId.systemDefault()).toLocalDate()
+    this.toLocalDateTime(TimeZone.currentSystemDefault()).date
 
 fun isApril1st(): Boolean {
-    val now = LocalDate.now()
+    val now = systemTimeNow().toLocalDate()
     return now.dayOfMonth == 1 && now.month == Month.APRIL
 }
+
+fun nowAsEpochMilliseconds(): Long = systemTimeNow().toEpochMilliseconds()
+
+fun systemTimeNow(): Instant = Clock.System.now()
+
+fun LocalDateTime.Companion.now(): LocalDateTime =
+    systemTimeNow().toLocalDateTime(TimeZone.currentSystemDefault())
+
+/** https://github.com/Kotlin/kotlinx-datetime#date--time-arithmetic */
+fun LocalDateTime.minusInSysTZ(value: Long, unit: DateTimeUnit): LocalDateTime =
+    TimeZone.currentSystemDefault().let { tz ->
+        this.toInstant(tz).minus(value, unit, tz).toLocalDateTime(tz)
+    }
+
+/** https://github.com/Kotlin/kotlinx-datetime#date--time-arithmetic */
+fun LocalDateTime.plusInSysTZ(value: Long, unit: DateTimeUnit): LocalDateTime =
+    TimeZone.currentSystemDefault().let { tz ->
+        this.toInstant(tz).plus(value, unit, tz).toLocalDateTime(tz)
+    }
```

Comment:
```suggestion
fun LocalDateTime.minusInSysTZ(value: Long, unit: DateTimeUnit): LocalDateTime {
    val tz = TimeZone.currentSystemDefault()
    return toInstant(tz).minus(value, unit, tz).toLocalDateTime(tz)
}

/** https://github.com/Kotlin/kotlinx-datetime#date--time-arithmetic */
fun LocalDateTime.plusInSysTZ(value: Long, unit: DateTimeUnit): LocalDateTime {
    val tz = TimeZone.currentSystemDefault()
    return toInstant(tz).plus(value, unit, tz).toLocalDateTime(tz)
}
```

I find this easier to read. Also, I'd suggest to either drop the `InSysTZ` in the function name (the doc comment should be enough of a warning) or write it out to `InSystemTimeZone`.

Reply:
https://github.com/streetcomplete/StreetComplete/pull/4471/commits/f77366a26d0e4c2e7a55efb94fe6848201e08455